### PR TITLE
Adding GitHub Action (in progress) and two test cases in C

### PR DESCRIPTION
**Pull request by CC-734**
the workflow is not totally working yet but runs when a file is pushed and will use diff to compare expected lex values to that of the output from lex.py

### CODE REVIEW

**Comment by adityaxdiwakar** [comment]
Until your PR is ready, you can let this PR stay as a draft. I'll review in the meantime, though.

**Comment by adityaxdiwakar** [comment]
What is not working?

File: tests/test1.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        max = x;
+    } else {
+        max = y;
+    }
+    printf("%d",max);
+}
+
+/*
+DIRECTIVE
+STR
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
STR(stdio.h)
```

File: tests/test1.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+INT_WORD
+IDENT(main)
+LBRACK
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
IDENT(main)
LPAREN
RPAREN
LBRACK
```

File: tests/test1.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+INT_WORD
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
INT_WORD
```

File: tests/test1.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+IDENT(x)
+SEMI
+INT_WORD
+IDENT(y)
+SEMI
+INT_WORD
+IDENT(max)
+SEMI
+
+IDENT(scanf)
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
SEMI
IDENT(scanf)
```

File: tests/test1.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+LPAREN
+STR
+COMMA
+IDENT(x)
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
COMMA
ANDREF
IDENT(x)
```

File: tests/test1.c [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+COMMA
+IDENT(y)
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
COMMA
ANDREF
IDENT(y)
```

File: tests/test1.c [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+RPAREN
+SEMI
+
+IF
+LPAREN
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
SEMI
IF
LPAREN
```

File: tests/test1.c [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+IDENT(x)
+GT
+IDENT(y)
+RBRACK
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
IDENT(y)
RPAREN
LBRACK
```

File: tests/test1.c [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+IDENT(max)
+EQUALS
+IDENT(x)
+SEMI
+RBRACK
+
+ELSE
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
RBRACK
ELSE
```

Comment (adityaxdiwakar) [reply tests/test1.c]: Wait... did I forget to implement else keyword? I don't think `py-lexer` knows this keyboard. Oops!

File: tests/test1.c [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+LBRACK
+IDENT(max)
+EQUALS
+IDENT(y)
+SEMI
+RBRACK
+
+IDENT(printf)
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
RBRACK
IDENT(printf)
LPAREN
```

File: tests/test1.c [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+STR
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
RBRACK

IDENT(printf)
STR(%d)
```

File: tests/test1.c [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+COMMA
+IDENT(max)
+SEMI
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
IDENT(max)
RPAREN
SEMI
```

File: tests/test1.c [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+RBRACK
```
Comment (adityaxdiwakar) [inline tests/test1.c]:
```suggestion
RBRACK
```

File: tests/test2.c [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+int main() {
+    float a = 10.0;
+    float b = 4.0;
+    return a / b;
+}
+
+/*
+DIRECTIVE
+STR
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
STR(stdio.h)
```

File: tests/test2.c [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+INT_WORD
+IDENT(main)
+LBRACK
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
IDENT(main)
LPAREN
RPAREN
LBRACK
```

File: tests/test2.c [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+FLOAT_WORD
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
FLOAT_WORD
```

File: tests/test2.c [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+IDENT(a)
+EQUALS
+FLOAT
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
FLOAT(10.0)
```

File: tests/test2.c [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+SEMI
+
+FLOAT_WORD
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
SEMI
FLOAT_WORD
```

File: tests/test2.c [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+IDENT(b)
+EQUALS
+FLOAT
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
FLOAT(4.0)
```

File: tests/test2.c [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+SEMI
+
+RETURN
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
SEMI
RETURN
```

File: tests/test2.c [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+IDENT(a)
+DIV
+IDENT(b)
+SEMI
+
+RBRACK
+
```
Comment (adityaxdiwakar) [inline tests/test2.c]:
```suggestion
SEMI
RBRACK
```

**Review by adityaxdiwakar — CHANGES_REQUESTED** [review]
Mostly good, missing some things -- I tried to leave most things as suggestions so you should be able to instantly commit without editing the files. There are **22** changes I suggest.

Beyond that, please separate out the expected output like I said in @ritvik-knagaram's PR in #1 

Thank you!

We'll talk offline about the workflow not working.

File: .github/workflows/run_py-lexer.yml [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+                    
+          for file in "${file_array[@]}"; do
+            file_extension="${file##*.}"
+            echo "Committed file: $file, Extension: $file_extension"
+            cat $file.$file_extension
+            python3 py.lex > committed_output.txt
+            
+            gcc $file.$file_extension
+            ./a.out > expected_output.txt
+
```
Comment (adityaxdiwakar) [inline .github/workflows/run_py-lexer.yml:49]:
```suggestion
```

File: .github/workflows/run_py-lexer.yml [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+            
+            diff committed_output.txt expected_output.txt > output_diff.txt || true
+            if [ -s output_diff.txt ]; then
+              echo "Outputs are different"
+              cat output_diff.txt
+              exit 1
+            else
+              echo "Outputs are identical"
+            fi
+          done
```
Comment (adityaxdiwakar) [inline .github/workflows/run_py-lexer.yml]:
```suggestion       
          for file in tests/*
          do
            echo "Testing $file against lexer"
            if [ -f "$file" ]; then
              name="${file%.*}"
              cat $file | python3 lex.py > output
              
              diff output "expected/$name.out" > output_diff
              status = $?
              if [ status -eq 0]; then
                echo "OK."
              fi
              else
                echo "Failed. See diff below."
                cat output_diff
                exit 1
            fi
          done
```

**Comment by adityaxdiwakar** [comment]
I fixed the workflow, now @CC-734 just needs to fix the tests (see previous comments) and we're good! :D

File: tests/test1.c (hunk 9 above)
Comment (CC-734) [reply tests/test1.c]: i changed it to indent(else) so it would pass the test but ELSE keyword would prob be nice

**Comment by adityaxdiwakar** [comment]
Please leave it as ELSE. This is a bug with the lexer. It's not currently
marked as a reserved keyword.

**Review by adityaxdiwakar — APPROVED** [review]

**Pull request merged**